hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -455,44 +459,47 @@ public void removeTrigger(String tenantId, String triggerId) {
     }
 
     private void removeTrigger(Trigger trigger) {
+        final String tenantId = trigger.getTenantId();
+        final String triggerId = trigger.getId();
+
+        // If necessary, clean up working memory
```

Comment:
We had an issue basing this work on the engine state.  In ConditionMatch.drl#SetFiringModeOnSatisfiedDampening we retract the trigger.  Now that the trigger fact is gone we would not go into this conditional and it would leave orphaned facts in the rulebase as well as not execute the cache work of the update of 'missingStates'.

So what I've done in this PR is to treat Trigger removal (via api) and Trigger retraction (via rulebase) independently.  Now, with trigger retraction there are the new rules to clean up the orphaned dampening, condition, and missingState facts, so e don't need to do this in the engine code.  With trigger removal, if the rulebase has the trigger fact, we do need to remove the rulebase facts via code, that is what this conditional is for.

Regardless of whether we are removing the trigger due to the rule-base (like handleAutoResolveTriggers) or API (e.g. REST),  I am *pretty sure* we want to do the alertsEngineCache  and missingStates work.  That is why it is now outside the 'if' block.  Does it make sense?